internal/policy/verify_test.go

```diff
@@ -0,0 +1,182 @@
+package policy
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ProtonMail/go-crypto/openpgp"
+	"github.com/adityasaky/gittuf/internal/gitinterface"
+	"github.com/adityasaky/gittuf/internal/rsl"
+	"github.com/adityasaky/gittuf/internal/signerverifier"
+	"github.com/adityasaky/gittuf/internal/signerverifier/dsse"
+	"github.com/adityasaky/gittuf/internal/tuf"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVerifyEntry(t *testing.T) {
+	repo, state := createTestRepository(t, createTestStateWithPolicy)
+
+	entry := rsl.NewEntry("refs/heads/main", plumbing.ZeroHash)
+	entryID := createTestRSLEntryCommit(t, repo, entry)
+	entry.ID = entryID
+
+	err := verifyEntry(context.Background(), repo, state, entry)
+	assert.Nil(t, err)
+}
+
+func createTestRSLEntryCommit(t *testing.T, repo *git.Repository, entry *rsl.Entry) plumbing.Hash {
```

Comment:
this entire test file is a mess with a lot of repetition from other methods that are private.